src/firefly/js/visualize/ui/ApiExpandedDisplay.jsx

```diff
@@ -33,8 +34,10 @@ export class ApiExpandedDisplay extends Component {
     }
 
     storeUpdate() {
-        if (visRoot()!==this.state.visRoot || lastMouseCtx() !==this.state.currMouseState) {
-            this.setState({visRoot:visRoot(), currMouseState:lastMouseCtx()});
+        if (visRoot()!==this.state.visRoot || 
+            lastMouseCtx() !==this.state.currMouseState || 
+            readoutRoot()!==this.state.readout) {
+            this.setState({visRoot:visRoot(), currMouseState:lastMouseCtx(), readout:readoutRoot()});
```

Comment:
Do functions (visRoot(), lastMouseCtx(), readoutRoot()) get called twice possibly if the code is written like this?  Sometime I was puzzled by  similar situation like this, not sure if this is a concern as thinking of execution time.